site/src/main/paradox/io/Tensorflow.md

```diff
@@ -0,0 +1,95 @@
+```
+
+## Writing
+
+Similar to reading, there are multiple ways to write Tensorflow files, depending on the format of the elements to be output.
+Each of these write methods is called `saveAsTfRecordFile`, but only one variant of the method is available based on the element type.
+
+* For `SCollection[T]` where `T` is a subclass of `Example`: @scaladoc[`saveAsTfRecordFile`](com.spotify.scio.tensorflow.syntax.ExampleSCollectionOps#saveAsTfRecordFile(path:String,suffix:String,compression:org.apache.beam.sdk.io.Compression,numShards:Int,shardNameTemplate:String,tempDirectory:String,filenamePolicySupplier:com.spotify.scio.util.FilenamePolicySupplier):com.spotify.scio.io.ClosedTap[org.tensorflow.proto.example.Example])
+* For `SCollection[Seq[T]]` where `T` is a subclass of `Example`: @scaladoc[`saveAsTfRecordFile`](com.spotify.scio.tensorflow.syntax.SeqExampleSCollectionOps#saveAsTfRecordFile(path:String,suffix:String,compression:org.apache.beam.sdk.io.Compression,numShards:Int,shardNameTemplate:String,tempDirectory:String,filenamePolicySupplier:com.spotify.scio.util.FilenamePolicySupplier):com.spotify.scio.io.ClosedTap[org.tensorflow.proto.example.Example])
+* For `SCollection[T]` where `T` is a subclass of `SequenceExample`: @scaladoc[`saveAsTfRecordFile`](com.spotify.scio.tensorflow.syntax.SequenceExampleSCollectionOps#saveAsTfRecordFile(path:String,suffix:String,compression:org.apache.beam.sdk.io.Compression,numShards:Int,shardNameTemplate:String,tempDirectory:String,filenamePolicySupplier:com.spotify.scio.util.FilenamePolicySupplier):com.spotify.scio.io.ClosedTap[org.tensorflow.proto.example.SequenceExample])
+* For `SCollection[Array[Byte]]`, where it is recommended that the bytes are a serialized `Example`:
+@scaladoc[`saveAsTfRecordFile`](com.spotify.scio.tensorflow.syntax.TFRecordSCollectionOps#saveAsTfRecordFile(path:String,suffix:String,compression:org.apache.beam.sdk.io.Compression,numShards:Int,shardNameTemplate:String,tempDirectory:String,filenamePolicySupplier:com.spotify.scio.util.FilenamePolicySupplier)(implicitev:T%3C:%3CArray[Byte]):com.spotify.scio.io.ClosedTap[Array[Byte]])
+
+```scala mdoc:compile-only
+import com.spotify.scio.values.SCollection
+import com.spotify.scio.tensorflow._
+import org.tensorflow.proto.example.{Example, SequenceExample}
+
+val recordBytes: SCollection[Array[Byte]] = ???
+val examples: SCollection[Example] = ???
+val seqExamples: SCollection[Seq[Example]] = ???
+val sequenceExamples: SCollection[SequenceExample] = ???
+
+recordBytes.saveAsTfRecordFile("gs://output-record-path")
+examples.saveAsTfRecordFile("gs://output-example-path")
+seqExamples.saveAsTfRecordFile("gs://output-seq-example-path")
+sequenceExamples.saveAsTfRecordFile("gs://output-sequence-example-path")
+```
+
+# Prediction/inference
+
+Scio supports preforming inference on a saved Tensorflow model.
+
+TODO I don't understand the difference between these
```

Comment:
Let's try to not merge that

Reply:
:) hoping somebody will help me out

Reply:
This is what I understood:

- `predict` is used with 'raw' model to identify input and output from the transformation and give tensor names.
- `predictWithSigDef` will use the embedded model's [SignatureDefs](https://www.tensorflow.org/tfx/serving/signature_defs#:~:text=A%20SignatureDef%20defines%20the%20signature%20of%20a%20computation,and%20can%20be%20specified%20when%20building%20a%20SavedModel.). input and output are identified by the name exported in the defs.